pygfx/controls/_base.py

```diff
@@ -0,0 +1,50 @@
+from typing import Tuple
+
+from ..linalg import Vector3
+from ..utils.viewport import Viewport
+
+
+# todo: I hate that these classes are plural, maybe rename to Controller?
+class BaseControls:
+    """Abstract base controls."""
+
+    def __init__(self):
+        pass
+
+    def update_camera(self, camera: "Camera") -> "BaseControls":
+        """Update the transform of the camera with the internal transform."""
+        rot, pos, zoom = self.get_view()
+        camera.rotation.copy(rot)
+        camera.position.copy(pos)
+        camera.zoom = zoom
+        return self
+
+    def add_default_event_handlers(self, viewport, camera):
+        """Apply the default interaction mechanism to a wgpu autogui canvas."""
+        viewport = Viewport.from_viewport_or_renderer(viewport)
+        viewport.renderer.add_event_handler(
+            lambda event: self.handle_event(event, viewport, camera),
+            "pointer_down",
+            "pointer_move",
+            "pointer_up",
+            "wheel",
+        )
+
+    def handle_event(self, event, viewport, camera):
+        pass
```

Comment:
why are the empty methods (`__init__` and `handle_event`) needed?

Reply:
Good point. In an earlier version the `__init__` wasn't empty, I will remove it, and raise notimplemented in the placeholder `handle_event`.